garrysmod/gamemodes/terrortown/gamemode/cl_scoring.lua

```diff
@@ -350,11 +371,40 @@ function CLSCORE:BuildHilitePanel(dpanel)
 end
 
 function CLSCORE:ShowPanel()
+   if IsValid(self.Panel) then
+      self:ClearPanel()
+   end
+
    local margin = 15
 
    local dpanel = vgui.Create("DFrame")
-   local w, h = 700, 500
-   dpanel:SetSize(700, 500)
+
+   local title
+   local starttime = self.StartTime
+   local endtime = starttime
+   local events = self.Events
+
+   for i = #events, 1, -1 do
+      local e = events[i]
+      if e.id == EVENT_FINISH then
+         endtime = e.t
+         title = self.WinTypes[e.win]
+         break
+      end
+   end
+
+   if title == nil then
+      title = self.WinTypes.Default
+   end
+
+   -- size the panel based on the win text w/ 88px horizontal padding and 44px veritcal padding
+   surface.SetFont("WinHuge")
+   local w, h = surface.GetTextSize(title.Text or self.WinTypes.Default.Text)

```

Comment:
Also, wouldn't this need to be calculated on the translated string instead of the translation identification string? E.g. `T(title.Text or self.WinTypes.Default.Text)` instead?

Reply:
I already fixed this in another diff but it appears I copied over an older version. Fixed now.